src/main/java/org/jenkinsci/remoting/util/AnonymousClassWarnings.java

```diff
@@ -51,16 +47,14 @@ public class AnonymousClassWarnings {
     private static final Logger LOGGER = Logger.getLogger(AnonymousClassWarnings.class.getName());
     private static final Map<Class<?>, Boolean> checked = new WeakHashMap<>();
 
-    private final static ExecutorService THREAD_POOL =  new AtmostOneThreadExecutor(new NamingThreadFactory(new DaemonThreadFactory(), AnonymousClassWarnings.class.getSimpleName()));
-
     /**
      * Checks a class which is being either serialized or deserialized.
      * A warning will only be printed once per class per JVM session.
      */
     public static void check(@NonNull Class<?> clazz) {
         synchronized (checked) {
-            if (checked.containsKey(clazz)) {
```

Comment:
Can we keep the containsKey check and move it above the sync block (double checked locking) to potentially lower contention since this seems like a hot path?

Reply:
Ah nevermind the map in question is WeakHashMap and not thread safe.